plugins/adrv9002.c

```diff
@@ -927,6 +1189,1082 @@ static void load_profile(GtkFileChooser *chooser, gpointer data)
+{
+	struct plugin_private *priv = data;
+	GtkLabel *info_label = GTK_LABEL(gtk_builder_get_object(priv->builder, "label_profile_debug"));
+	char message[BUFSIZ];
+
+	sprintf(message, "%s%s", gtk_label_get_text(info_label), g_locale_to_utf8(string, -1, NULL, NULL, NULL));
+	gtk_label_set_text(info_label, message);
+}
+
+static void profile_gen_set_debug_info(gpointer data, char *string)
+{
+	struct plugin_private *priv = data;
+	gtk_label_set_text(GTK_LABEL(gtk_builder_get_object(priv->builder, "label_profile_debug")),
+			   g_locale_to_utf8(string, -1, NULL, NULL, NULL));
+}
+
+static void profile_gen_save_type_changed(GtkComboBoxText* self, struct plugin_private *priv)
+{
+	// 0 - stream image
+	// 1 - profile
+	bool save_type = atoi(gtk_combo_box_get_active_id(GTK_COMBO_BOX(self)));
+
+	char filename[64];
+	sprintf(filename, "adrv9002%s", save_type? ".json": ".stream");
+	GtkFileChooser *chooser = GTK_FILE_CHOOSER(gtk_builder_get_object(priv->builder, "save_panel"));
+	gtk_file_chooser_set_current_name(chooser, filename);
+}
+
+static void profile_gen_save_dialog_show(GtkButton* self, gpointer data)
+{
+	struct plugin_private *priv = data;
+	GtkFileChooser *chooser = GTK_FILE_CHOOSER(gtk_builder_get_object(priv->builder, "save_panel"));
+
+	gtk_file_chooser_set_action(chooser, GTK_FILE_CHOOSER_ACTION_SAVE);
+	gtk_file_chooser_set_do_overwrite_confirmation(chooser, TRUE);
+
+	gtk_file_chooser_set_current_folder(chooser, getenv("HOME"));
+	profile_gen_save_type_changed(GTK_COMBO_BOX_TEXT(gtk_builder_get_object(priv->builder, "cb_save_type")), priv);
+
+	gtk_widget_show(GTK_WIDGET(chooser));
+}
+
+static void profile_gen_update_orx(GtkComboBox* self, struct plugin_private *priv)
+{
+	int chann;
+	char widget_str[25];
+	bool tdd_en = atoi(gtk_combo_box_get_active_id(GTK_COMBO_BOX(gtk_builder_get_object(priv->builder, "cb_radio_duplex")))) == 0;
+	bool tx_en;
+
+	for (chann = 0; chann < 2; chann ++) {
+		sprintf(widget_str, "cb_tx_chan%d_en", chann + 1);
+		tx_en = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)));
+
+		sprintf(widget_str, "frame_radio_orx%d", chann + 1);
+		gtk_widget_set_sensitive(GTK_WIDGET(gtk_builder_get_object(priv->builder, widget_str)), tx_en && tdd_en);
+
+		// deactivate if disabled
+		if (!(tx_en && tdd_en)) {
+			sprintf(widget_str, "cb_radio_orx%d_en", chann + 1);
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), false);
+		}
+	}
+}
+
+
+char *extract_value_between(const char *str, const char *begin, const char *end)
+{
+	// if end is empty it will stop at the end of str
+	const char *i1 = strstr(str, begin);
+	if(i1 != NULL)
+	{
+		const size_t pl1 = strlen(begin);
+		const char *i2 = strcmp(end, "") != 0? strstr(i1 + pl1, end): NULL;
+		const size_t mlen = strcmp(end, "") != 0? i2 - (i1 + pl1): (int)(strlen(i1) - pl1);
+		char *ret = malloc(mlen + 1);
```

Comment:
Since extract_value_between() returns content that is allocated on heap. Are you making sure things are getting freed?